src/iceberg_functions/iceberg_multi_file_reader.cpp

```diff
@@ -291,34 +339,9 @@ void IcebergMultiFileReader::FinalizeBind(MultiFileReaderData &reader_data, cons
 void IcebergMultiFileReader::ApplyEqualityDeletes(ClientContext &context, DataChunk &output_chunk,
                                                   const IcebergMultiFileList &multi_file_list,
                                                   const IcebergManifestEntry &data_file,
-                                                  const vector<MultiFileColumnDefinition> &local_columns) {
-	vector<reference<IcebergEqualityDeleteRow>> delete_rows;
-
-	auto &metadata = multi_file_list.GetMetadata();
-	auto delete_data_it = multi_file_list.equality_delete_data.upper_bound(data_file.sequence_number);
-	//! Look through all the equality delete files with a *higher* sequence number
-	for (; delete_data_it != multi_file_list.equality_delete_data.end(); delete_data_it++) {
-		auto &files = delete_data_it->second->files;
-		for (auto &file : files) {
-			auto &partition_spec = metadata.partition_specs.at(file.partition_spec_id);
-			if (partition_spec.IsPartitioned()) {
-				if (file.partition_spec_id != data_file.partition_spec_id) {
-					//! Not unpartitioned and the data does not share the same partition spec as the delete, skip the
-					//! delete file.
-					continue;
-				}
-				D_ASSERT(file.partition_values.size() == data_file.partition_values.size());
-				for (idx_t i = 0; i < file.partition_values.size(); i++) {
-					if (file.partition_values[i] != data_file.partition_values[i]) {
-						//! Same partition spec id, but the partitioning information doesn't match, delete file doesn't
-						//! apply.
-						continue;
-					}
-				}
-			}
-			delete_rows.insert(delete_rows.end(), file.rows.begin(), file.rows.end());
-		}
-	}
+                                                  const vector<MultiFileColumnDefinition> &local_columns,
+                                                  const unordered_map<idx_t, idx_t> &field_id_to_result_id) {
+	auto delete_rows = multi_file_list.GetEqualityDeletesForFile(data_file);
```

Comment:
I deduplicated your iteration, with the logic that was in `ApplyEqualityDeletes`, which was more complete.
Your iteration didn't take into account the partition spec and partitioning information, which I imagine was not missing on purpose (?)